src/project/projecterrors.h

```diff
@@ -46,6 +52,59 @@ inline muse::Ret make_ret(Err e)
 {
     return muse::Ret(static_cast<int>(e));
 }
+
+inline muse::Ret make_ret(Err err, const muse::io::path_t& filePath)
+{
+    auto ret = muse::Ret(static_cast<int>(err));
```

Comment:
please move this to cpp, in general, if the function is not a one-liner, like the one above, it should be in cpp. This makes headers easier to read.